### ISSUE TITLE
Error handling / example doesn't work

### ISSUE THREAD

**Issue by ChristianSchab** [Opteo/google-ads-api#28]
Hi guys,

I'm struggling to add a keyword. 
Even the example doesn't work for me.

My code:

```
const { GoogleAdsApi, enums } = require('google-ads-api')

// Make sure you pass in valid authentication details!
const client = new GoogleAdsApi({
    client_id: '79573XXXXXXXXXXXXXXXXXXXX3.apps.googleusercontent.com',
    client_secret: 'sGCJRDXXXXXXXXXXXXXXXXXXXXQN1pq4',
    developer_token: 'hquXXXXXXXXXXXXXXXXXXXXB3jzPg',
})

async function main() {
    const customer = client.Customer({
        customer_account_id: '111222333',
        login_customer_id : '4445556666', // MCC
        refresh_token: "1/oWh6IPXXXXXXXXXXXXXXXq4P1X-M"
    })

    // Bonus: If you're using Typescript, set the type here to "types.Keyword" for autocomplete!
    const keyword = {
        ad_group: 'customers/111222333/adGroups/78901234567',
        status: enums.AdGroupCriterionStatus.ENABLED,
        text: 'hotels london',
        match_type: enums.KeywordMatchType.EXACT,
    }

    try {
        const { results } = await customer.adGroupCriterion.create(keyword)
        /*
            The newly created ad group criterion will have a resource name in the following format:
            "customers/{customer_id}/adGroupCriteria/{ad_group_id}~{criterion_id}"
        */
    } catch (err) {
        if (err.code.adGroupCriterionError === enums.AdGroupCriterionError.INVALID_KEYWORD_TEXT) {
            console.log(`Keyword with text "${keyword.text}" is invalid!`)
        } else {
            console.log(err);
        }
    }
}

main()
```

My output:

```
{ Error: The required field was not present in the resource.
    at AdGroupCriterionService.<anonymous> (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/google-ads-api/build/services/service.js:142:23)
    at Generator.throw (<anonymous>)
    at rejected (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/google-ads-api/build/services/service.js:5:65)
    at <anonymous>
  code: 
   { requestError: 0,
     biddingStrategyError: 0,
     urlFieldError: 0,
     listOperationError: 0,
     queryError: 0,
     mutateError: 0,
     fieldMaskError: 0,
     authorizationError: 0,
     internalError: 0,
     quotaError: 0,
     adError: 0,
     adGroupError: 0,
     campaignBudgetError: 0,
     campaignError: 0,
     authenticationError: 0,
     adGroupCriterionError: 0,
     adCustomizerError: 0,
     adGroupAdError: 0,
     adSharingError: 0,
     adxError: 0,
     assetError: 0,
     biddingError: 0,
     campaignCriterionError: 0,
     collectionSizeError: 0,
     countryCodeError: 0,
     criterionError: 0,
     customerError: 0,
     dateError: 0,
     dateRangeError: 0,
     distinctError: 0,
     feedAttributeReferenceError: 0,
     functionError: 0,
     functionParsingError: 0,
     idError: 0,
     imageError: 0,
     languageCodeError: 0,
     mediaBundleError: 0,
     mediaFileError: 0,
     multiplierError: 0,
     newResourceCreationError: 0,
     notEmptyError: 0,
     nullError: 0,
     operatorError: 0,
     rangeError: 0,
     recommendationError: 0,
     regionCodeError: 0,
     settingError: 0,
     stringFormatError: 0,
     stringLengthError: 0,
     operationAccessDeniedError: 0,
     resourceAccessDeniedError: 0,
     resourceCountLimitExceededError: 0,
     adGroupBidModifierError: 0,
     contextError: 0,
     fieldError: 2,
     sharedSetError: 0,
     sharedCriterionError: 0,
     campaignSharedSetError: 0,
     conversionActionError: 0,
     conversionAdjustmentUploadError: 0,
     conversionUploadError: 0,
     headerError: 0,
     databaseError: 0,
     policyFindingError: 0,
     enumError: 0,
     keywordPlanError: 0,
     keywordPlanCampaignError: 0,
     keywordPlanNegativeKeywordError: 0,
     keywordPlanAdGroupError: 0,
     keywordPlanKeywordError: 0,
     keywordPlanIdeaError: 0,
     accountBudgetProposalError: 0,
     userListError: 0,
     changeStatusError: 0,
     feedError: 0,
     geoTargetConstantSuggestionError: 0,
     feedItemError: 0,
     labelError: 0,
     billingSetupError: 0,
     customerClientLinkError: 0,
     customerManagerLinkError: 0,
     feedMappingError: 0,
     customerFeedError: 0,
     adGroupFeedError: 0,
     campaignFeedError: 0,
     customInterestError: 0,
     extensionFeedItemError: 0,
     adParameterError: 0,
     feedItemValidationError: 0,
     extensionSettingError: 0,
     feedItemTargetError: 0,
     policyViolationError: 0,
     mutateJobError: 0,
     partialFailureError: 0,
     policyValidationParameterError: 0 },
  request: 
   { customerId: '8849161347',
     operationsList: [ [Object] ],
     partialFailure: false,
     validateOnly: false },
  request_id: 'P0zzq9JTK3piHQocRGo5zg',
  failure: 
   { Error: 3 INVALID_ARGUMENT: The required field was not present in the resource.
    at Object.exports.createStatusError (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/grpc/src/common.js:91:15)
    at Object.onReceiveStatus (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/grpc/src/client_interceptors.js:1204:28)
    at InterceptingListener._callNext (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/grpc/src/client_interceptors.js:568:42)
    at InterceptingListener.onReceiveStatus (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/grpc/src/client_interceptors.js:618:8)
    at Object.grpc_1.default.ListenerBuilder.withOnReceiveStatus [as onReceiveStatus] (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/google-ads-node/build/lib/interceptor.js:68:17)
    at InterceptingListener._callNext (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/grpc/src/client_interceptors.js:568:42)
    at InterceptingListener.onReceiveStatus (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/grpc/src/client_interceptors.js:618:8)
    at Object.grpc_1.default.ListenerBuilder.withOnReceiveStatus [as onReceiveStatus] (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/google-ads-node/build/lib/interceptor.js:151:13)
    at InterceptingListener._callNext (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/grpc/src/client_interceptors.js:568:42)
    at InterceptingListener.onReceiveStatus (/Users/<USER>/Development/WebstormProjects/<PROJECT>/node_modules/grpc/src/client_interceptors.js:618:8)
     code: 3,
     metadata: Metadata { _internal_repr: [Object] },
     details: 'The required field was not present in the resource.' } }

```

How to get a helpful error message?
Just received 
> Error: 3 INVALID_ARGUMENT: The required field was not present in the resource.

So which required field is missing?

BTW: If I comment out the status row

```
const keyword = {
        ad_group: 'customers/8849161347/adGroups/68449099820',
        // status: enums.AdGroupCriterionStatus.ENABLED,
        text: 'hotels london',
        match_type: enums.KeywordMatchType.EXACT,
    }
```

The example works. The keyword is created.

I'm looking forward to any hint.

Thanks a lot.

Best,
Christian

**Comment by mateuszjanusz** [comment]
Hi @ChristianSchab,

I tried to run the example and I can confirm that I also receive the same error message. 

This is because we pass the keyword object in an incorrect format. That object should look like this:
```javascript
const keyword = {
     ad_group: 'customers/8849161347/adGroups/68449099820',
     status: enums.AdGroupCriterionStatus.ENABLED,
     keyword: {
          text: 'hotels london',
          match_type: enums.KeywordMatchType.EXACT,
     }
}
```

By running checks, we also discovered a bug in the low-level library (https://github.com/Opteo/google-ads-node/pull/8). This only happens when you pass more than one enum in the create object (this explains why you did not receive the same error when you commented out the status row). 

We're planning to release that fix and a new version of the library today.
I'm also going to update the incorrect example.

Thanks for reporting this!
